main.cpp

```diff
@@ -38,63 +38,61 @@ struct CompareTValues                                //4
         if(a != nullptr && b != nullptr)
         {
             if( a->value < b->value ) return a;
-            if( a->value > b->value ) return b;
+            else if( a->value > b->value ) return b;
+            else
+            {
+                std::cout << "Values are equal, returning first arg..." << std::endl;
+                return a;
+            }
         }
+        std::cout << "got a nullptr, returning nullptr" << std::endl;
         return nullptr;
     }
```

Comment:
This is still not correct.
Ask yourself:
1) can the return type be converted to a reference?  Hint: what do the instructions say?
2) can the function arguments be converted to a reference? 
- [ ] revise